lib/faker/gov/it.ex

```diff
@@ -0,0 +1,161 @@
+defmodule Faker.Gov.It do
+  @moduledoc """
+  Generating Italian fiscal codes
+  """
+
+  require Integer
+
+  alias Faker.Util
+
+  # @doc """
+  # Returns a random Italian fiscal code
+
+  # ## Examples
+
+  #     iex> Faker.Gov.It.fiscal_id
+  #     "ZRSNND48A23E151U"
+  #     iex> Faker.Gov.It.fiscal_id
+  #     "CCPWLC58L04I289K"
+  #     iex> Faker.Gov.It.fiscal_id
+  #     "TTKRQQ77P43I477X"
+  # """
+  @spec fiscal_id() :: binary()
+  def fiscal_id do
+    surname()
+    |> Kernel.<>(name())
+    |> Kernel.<>(birth_year())
+    |> Kernel.<>(birth_month())
+    |> Kernel.<>(birth_day())
+    |> Kernel.<>(town_code_letter())
+    |> Kernel.<>(town_code_numbers())
+    |> (&Kernel.<>(&1, cin(&1))).()
```

Comment:
this line concerns me a bit. what do you say if we move all but this into separate function and use it as:

```elixir
def fiscal_id do
  a = your_new_function()
  a <> cin(a)
end
```

actually, now I'm writing this and I don't like it also.

Any thoughts @elixirs/faker ?

Reply:
Yeah I don't like it either... but it's the only function that takes the whole input as first param since it's basically a CRC. Open to suggestions!